detectable/src/main/java/com/synopsys/integration/detectable/detectables/pnpm/lockfile/process/PnpmLockYamlParser.java

```diff
@@ -4,120 +4,73 @@
 import java.io.FileNotFoundException;
 import java.io.FileReader;
 import java.io.IOException;
-import java.util.Collections;
-import java.util.LinkedList;
 import java.util.List;
-import java.util.Map;
 
-import org.apache.commons.collections4.MapUtils;
-import org.apache.commons.collections4.Predicate;
-import org.apache.commons.lang3.StringUtils;
 import org.jetbrains.annotations.Nullable;
 import org.yaml.snakeyaml.DumperOptions;
 import org.yaml.snakeyaml.LoaderOptions;
 import org.yaml.snakeyaml.Yaml;
 import org.yaml.snakeyaml.constructor.Constructor;
+import org.yaml.snakeyaml.constructor.ConstructorException;
 import org.yaml.snakeyaml.representer.Representer;
 
 import com.synopsys.integration.detectable.detectable.codelocation.CodeLocation;
+import com.synopsys.integration.detectable.detectable.util.EnumListFilter;
+import com.synopsys.integration.detectable.detectables.pnpm.lockfile.model.PnpmDependencyType;
 import com.synopsys.integration.detectable.detectables.pnpm.lockfile.model.PnpmLockYaml;
-import com.synopsys.integration.detectable.detectables.pnpm.lockfile.model.PnpmProjectPackage;
+import com.synopsys.integration.detectable.detectables.pnpm.lockfile.model.PnpmLockYamlv5;
+import com.synopsys.integration.detectable.detectables.pnpm.lockfile.model.PnpmLockYamlv6;
 import com.synopsys.integration.exception.IntegrationException;
 import com.synopsys.integration.util.NameVersion;
 
 public class PnpmLockYamlParser {
```

Comment:
This class now handles the high level parsing of the pnpm-lock.yaml file. The large "deletions" in this class are moved to the v5 PnpmLockYamlParserv5 so that the code operates as before. New additions are made to the v6 PnpmLockYamlParserv6 to account for the changes in v6 and general improvements (see PR description).